### PR TITLE
mobile: fix onboarding

### LINKED ISSUE #853: Onboarding issues
- Clicking invite link and deep linking into onboarding does not skip invite screen
- User: "the app asks for copy paste permission but doesn't show it being used, it's a bit confusing, I was trying to manually paste it to the account name field"
    - we should make status on username page something like "✅valid invite from <>" on load
- payment links are not auto-claimed on account creation, user starts with 0 balance
    - "2024-03-14T18:10:28.165Z ERR [ACCOUNT] error claiming invite payment link: Error: Cannot claim without seed"

### PR DESCRIPTION
**Pull request by dcposch**
fixes #853

Screenshot 2024-03-14 at 23 31 55
Screenshot 2024-03-14 at 23 32 05
Screenshot 2024-03-15 at 00 29 46
Screenshot 2024-03-15 at 01 35 44
image
image

### CODE REVIEW

**Review by nalinbhardwaj — APPROVED** [review]
lgtmit

File: apps/daimo-mobile/src/common/nav.ts [hunk 1]
```diff
@@ -146,6 +154,45 @@ export function useOnboardingNav<
...
+    await handleOnboardingDeepLink(nav, deeplink);
+    markInitialDeepLinkHandled();
+  };
+
+  useEffect(() => {
+    navToInitialInviteDeepLinkIfPresent();
+    const listener: URLListener = ({ url }) =>
+      handleOnboardingDeepLink(nav, url);
+    const sub = addEventListener("url", listener);
+    return () => sub.remove();
```
Comment (nalinbhardwaj) [inline apps/daimo-mobile/src/common/nav.ts:177]: just to double check this doesn’t break deep links if you switch tabs to another one right?

File: apps/daimo-mobile/src/view/screen/HomeScreen.tsx [hunk 2]
```diff
@@ -303,12 +303,13 @@ function useInitNavLinks() {
...
     deepLinkInitialised = true;
-    getInitialURLOrTag(false).then((url) => {
+    getInitialDeepLink().then((url) => {
       if (url == null) return;
       handleDeepLink(nav, url);
     });
 
-    addEventListener("url", ({ url }) => handleDeepLink(nav, url));
+    const sub = addEventListener("url", ({ url }) => handleDeepLink(nav, url));
+    return () => sub.remove();
```
Comment (nalinbhardwaj) [inline apps/daimo-mobile/src/view/screen/HomeScreen.tsx:312]: just to double check this doesn’t break deep links if you switch tabs to another one right?

Comment (dcposch) [reply apps/daimo-mobile/src/view/screen/HomeScreen.tsx:312]: just tested--works

File: apps/daimo-mobile/src/logic/linkStatus.ts [hunk 3]
```diff
@@ -1,14 +1,29 @@
...
-  const ret = rpcHook.getLinkStatus.useQuery({ url: url! }, { enabled: !!url });
-  return ret;
+  const sanitizedLink = link && stripSeedFromNoteLink(link);
+  const url = sanitizedLink && formatDaimoLink(sanitizedLink);
+  const { rpcHook } = env(daimoChain);
+  return rpcHook.getLinkStatus.useQuery({ url: url! }, { enabled: !!url });
+}
+
+export async function fetchInviteLinkStatus(inviteLink: DaimoLink | undefined) {
+  const daimoChain = getAccountManager().getDaimoChain();
```
Comment (nalinbhardwaj) [inline apps/daimo-mobile/src/logic/linkStatus.ts:23]: could pass as arg to fix failing test import

Comment (dcposch) [reply apps/daimo-mobile/src/logic/linkStatus.ts:23]: will do. as part of #620 we can separate testnet vs mainnet links onto separate domains (> we already have test.daimo.xyz ) and simplify a lot of code

**Pull request merged**